src/components/timeline/TrackCustomMarkerGraph.js

```diff
@@ -302,19 +310,24 @@ class TrackCustomMarkerCanvas extends React.PureComponent<CanvasProps> {
             ctx.fillStyle = ctx.strokeStyle;
 
             for (let i = sampleStart; i < sampleEnd; i++) {
-              const marker = getMarker(collectedSamples.markerIndexes[i]);
-              const x = Math.round(
-                (marker.start - rangeStart) * millisecondWidth
-              );
+              let marker = getMarker(collectedSamples.markerIndexes[i]);
+              const x = getX(marker);
+              let y = getY(i);
+
+              // If we have multiple markers to draw on the same horizontal pixel,
+              // draw only the one with the maximum value to save time.
+              while (i + 1 < sampleEnd) {
+                const nextMarker = getMarker(
+                  collectedSamples.markerIndexes[i + 1]
+                );
+                if (getX(nextMarker) !== x) {
+                  break;
+                }
+
+                marker = nextMarker;
+                y = Math.min(y, getY(++i));
```

Comment:
Don't we want to get the max value with the Math.max here instead of min?

Reply:
We want the max value, which (unless I'm confused) is the min y (as y = 0 would mean top of the canvas).

Reply:
Oh welp right, I was the one who's confused not you. Sorry about the confusion!